### PR TITLE
Rearange structure of this projec to not expose a public API

### PR DESCRIPTION
**Pull request by cbourjau**
Some parts of this project may be useful to other cargo projects, but
if we expose an API we should document it properly and also make some
concious commitments to its stability. Thus, for now I removed the
`lib` part completly, leaving this repository as `bin`-only.

Furthermore, cleaned up (simplified) some thing in the assambly of the
inferior command.

### CODE REVIEW

File: src/runner.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    Ok(())
+}
+
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_build_ops() {
```
Comment (barskern) [inline src/runner.rs]: Maybe move this to the `cargo.rs` file as it's testing something specifically in that file?

**Review by barskern — CHANGES_REQUESTED** [review]
I like the addition of `--print-only`! A "hack" to do this previously would be `cargo with echo -- run`.

EDIT; Forgot to mention that I think the change to remove the lib part is smart.

File: src/runner.rs (hunk 1 above)
Comment (barskern) [reply src/runner.rs]: This will also resolve the build error.

Comment (cbourjau) [reply src/runner.rs]: Ups! that `--print-only` was not supposed to be there (yet). Currently, there is no functionality triggered by that flag, yet, so I removed it again from this PR.

Comment (cbourjau) [reply src/runner.rs]: I also ran `clippy`

Comment (barskern) [reply src/runner.rs]: Good thing I commented on it. I thought it was a work-in-process and that you would commit more to this branch. :joy:

**Comment by barskern** [comment]
Looks good to me!

**Review by barskern — APPROVED** [review]

**Pull request merged**